internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -34,16 +34,30 @@
 package clustermanager
 
 import (
+	"encoding/json"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
+	mm "github.com/Azure/azure-storage-fuse/v2/internal/dcache/metadata_manager"
 )
 
 type ClusterManagerImpl struct {
 	storageCallback dcache.StorageCallbacks
+	metaDataManager mm.MetadataManager
+	storagePath     string
 }
 
 // Start implements ClusterManager.
-func (c *ClusterManagerImpl) Start(*dcache.DCacheConfig, []dcache.RawVolume) error {
-	//create clusterMap.json file
+func (cmi *ClusterManagerImpl) Start(dCacheConfig *dcache.DCacheConfig, rvs []dcache.RawVolume) error {
+	cmi.storagePath = "__CACHE__" + dCacheConfig.CacheId
+	cmi.metaDataManager, _ = mm.NewMetadataManager(cmi.storagePath)
```

Comment:
metadatamanager should not be taking cmi.storagePath  as argument.
Being a low level component, and it being incharge of metadata, must manage the md itself, including setting the md root (as I mentioned in the other PR). That would also be inline with the fact that metadatamanager can be changed easily.
Infact we should not have to call NewMetadataManager() to create one, it should automatically get created from init() inside metadata manager package and then all calls should be like mm.cmi.metaDataManager.CreateInitialClusterMap() and not like cmi.metaDataManager.CreateInitialClusterMap().

That makes metadatamanager a singleton, which it is.